### PR TITLE
Move over initial hardware backend

### PR DESCRIPTION
**Pull request by ayush-goyal**
### Description

- Moved over hardware backend from https://github.com/HackGT/bolt with Prisma backend
- Converted GraphQL api to REST api with very rough routes
- There are definitely some bugs, but wanted to create some initial structure

### CODE REVIEW

File: services/hardware/src/routes/index.ts [hunk 1]
```diff
@@ -5,10 +5,12 @@ import { categoryRouter } from "./category";
...
 import { hardwareRequestRouter } from "./hardware-request";
+import { userRoutes } from "./user";
 
 export const defaultRouter = express.Router();
 
 defaultRouter.use("/items", itemRouter);
 defaultRouter.use("/categories", categoryRouter);
 defaultRouter.use("/locations", locationRouter);
 defaultRouter.use("/hardware-requests", hardwareRequestRouter);
+defaultRouter.use("/users", userRoutes);
```
Comment (Roshan-D) [inline services/hardware/src/routes/index.ts:16]: i think you forgot the hardware settings routes here

File: services/hardware/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  location         Location  @relation(fields: [locationId], references: [id])
+  requests         Request[]
+
+  categoryId Int
+  locationId Int @default(1)
+
+  @@map("item")
+}
+
+model Location {
```
Comment (Roshan-D) [inline services/hardware/prisma/schema.prisma:54]: see my slack message

**Review by Roshan-D — CHANGES_REQUESTED** [review]
looks good for the most part, couple small changes I commented about

File: services/hardware/src/routes/index.ts (hunk 1 above)
Comment (ayush-goyal) [reply services/hardware/src/routes/index.ts:17]: Added it in

**Review by Roshan-D — APPROVED** [review]
Fire!

**Review by thaneesh-babu — APPROVED** [review]

**Pull request merged**